stdlib/asm/math/ext2.masm

```diff
@@ -51,4 +51,41 @@ export.sub
     movdn.2     #[a1,b1,a0-b0,...]
     swap        #[b1,a1,a0-b0,...]
     sub         #[a1-b1,a0-b0,...]
-end
\ No newline at end of file
+end
+
+#! Given an invertible quadratic extension field element on stack, this routine computes 
+#! multiplicative inverse of that element, using non-deterministic technique ( i.e. takes help
+#! of advice provider ). Just to ensure that non-deterministic computation resulted in corrected value,
+#! it multiplies input operand with computed output, over quadratic extension field which must produce
+#! multiplicative identity (1, 0) of quadratic extension field.
+#!
+#! Expected input stack
+#!
+#! [a1, a0, ...] | a = (a0, a1) ∈ Quadratic extension field over F_q, q = 2^64 - 2^32 + 1
+#!
+#! Expected output stack
+#!
+#! [a'1, a'0, ...] | a' = (a'0, a'1) ∈ Quadratic extension field over F_q, q = 2^64 - 2^32 + 1
+#!
+#! Following is what is checked after reading result of computation, performed outside of VM
+#!
+#! a  = (a0, a1)
+#! a' = (a'0, a'1) ( = a ^ -1 )
+#!
+#! b  = a * a' ( mod P ) | P = irreducible polynomial x^2 - x + 2 over F_q, q = 2^64 - 2^32 + 1
+#! assert b  = (1, 0) | (1, 0) is the multiplicative identity of extension field
+export.inv
+    adv.ext2inv
+    adv_push.2
+    dup.1
+    dup.1
+
+    movup.5
+    movup.5
+    exec.mul
+
+    eq.0
+    assert
+    eq.1
+    assert
```

Comment:
Missed this instruction, fixed in 96f9f173c29264793f18fe7f9ebcc4a2363185d1. 

> **Note** It does save us 1 -cycle overall, because `assertz` costs us 2 -cycles.

Reply:
I'm surprised since `assertz` replaced 2 instructions and here `eq.1` should actually have taken up 2 instructions - so, I expected that by eliminating it we'd get back 2 cycles.

Reply:
I'm looking at https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/field_operations.html#comparison-operations, in says `eq.1` can take between (1-2) -cycles. It's probably the case where it's taking 1 -cycle. Though I'd expect it to take 2 -cycles, because first it should push `1` onto stack and then only it can issue `eq` instruction. 

Though probably I'm not considering something else.